src/Sulu/Component/SmartContent/ContentType.php

```diff
@@ -379,16 +380,21 @@ private function getProvider(PropertyInterface $property)
      * @param string $pageParameter
      *
      * @return int
+     *
+     * @throws PageOutOfBoundsException
      */
     private function getCurrentPage($pageParameter)
     {
-        if ($this->requestStack->getCurrentRequest() !== null) {
-            $page = $this->requestStack->getCurrentRequest()->get($pageParameter, 1);
-        } else {
-            $page = 1;
+        if ($this->requestStack->getCurrentRequest() === null) {
+            return 1;
+        }
+
+        $page = $this->requestStack->getCurrentRequest()->get($pageParameter, 1);
+        if ($page < 1 || $page > PHP_INT_MAX) {
```

Comment:
that would be a BC break in my opinion - because if someone uses the smart-content without checking for `hasNextPage` it would throw an exception now. @chirimoya what do you think?